pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -28,6 +29,9 @@ var (
 		Author:      "",
 		Company:     "",
 	}
+
+	// nextDB is used to provide a unique name so there is not false sharing between in-memory sqlite3 databases
+	nextDB uint64
```

Comment:
Can you help me understand what the deal is with this bit?

Not sure if it's possible, but it would be great if we could tuck away the sqlite-specific stuff in `sqlite.go` instead of here

Reply:
The comment is worded poorly, i will reword it.

In a nutshell, the code previously opened `'":memory:"`, which provides a completely private in-memory database. Unfortunately, that does not work well with golang sql, since any operation performed over a new database connection being opened (e.g., a transaction) gets a fresh database.  You can use the same in-memory database by connecting to a database with a regular uri name with `mode=memory`. However, if you ever want to run tests in parallel, you really need a distinct database, which can be accomplished by using a different name in the uri, hence the `nextDB` id. 

See https://www.sqlite.org/inmemorydb.html for details.